mitodl/teachersportal
# Added product detail API to replace ccxcon API

#### What's this PR do?
- Removes /ccxcon/v1/ API
- Adds a detail view for the REST API /api/v1/products/upc/, where the upc is the product type combined with the uuid like this: Course_uuid. 
- Changes the product API results to include an 'info' value with information retrieved from CCXCon. This should only be present in the product API detail view, the list view should have this be `null`
- Other fields added to the product API results. `parent_external_pk` is removed and replaced with `parent_upc`.
- Actions, reducers and related functions are changed to use the product detail API.
- Python tests have been refactored to test smaller units
#### Where should the reviewer start?

`portal/views.py` and `static/js/containers/CourseDetailPage.js`
#### How should this be manually tested?

Add a product to the database:
- Start teacher's portal locally and go to http://localhost:8075/oscar/dashboard/catalogue/
- Under 'Create new product of type' choose 'Course' and click the plus next to it
- Put anything you want for title. UPC **must** begin with `Course_` but otherwise can be anything
- On the left click 'Categories' and then select 'Course' which should be the only category available
- Click 'Save and continue editing'
- On the left click 'Variants', then click 'Add variant'
- Fill in whatever title you want. The UPC **must** start with `Module_` but otherwise can be anything
- On the left click 'Stock and Pricing'. Partner should be edX (the only partner which should be available). SKU _must_ be exactly the same as the value you put in the UPC text field (the one that began with `Module_`). Currency should be `$`. Enter a number for "Price (excl tax)" (make sure it's not "Cost price" or something else). Click Save.
- Go to `/api/v1/products/` and verify that you see two products, the course and then the module. It must be at the end of the list since this list should be sorted by creation order.

At this point, it should be enough to log in and view a product page and make sure it shows up like it did before. Also make sure that no text is missing and the correct number of modules show up in the list.
#### Any background context you want to provide?

NOTE: this PR is rebased off of the branch for #135

This should also finish the authorization work for TP. Users who are not logged in should only be able to access certain APIs like login, registration, activation, and not products.
#### What are the relevant tickets?

Fixes #129
